src/main/java/frc/robot/commands/drive/Drive.java

```diff
@@ -15,24 +17,88 @@
      */
     private boolean sideFlipped = false;
 
+    private double pitchDegrees;
+
     public Drive() {
         requires(Drivetrain.getInstance());
     }
 
+    public static double powerCurve(double x) {
+        // https://www.desmos.com/calculator/g07ukjj7bl
+        double curve = (0.5D * (Math.atan(Math.PI * (Math.abs(x) - 0.5D)))) + 0.5D;
+        return Math.copySign(curve, x);
+    }
+
+    /**@param array array of motor power values
+     * If any of the values are more than 1, they aren't valid values for motor power.
+     * If so, it divides all array values by the largest value to preserve the value ratios while making them valid motor power values.
+    */
+    public static void normalize(double[] array) {
+        double max = Math.abs(array[0]);
+        boolean normFlag = max > 1;
+
+        for (int i = 1; i < array.length; i++) {
+            if (Math.abs(array[i]) > max) {
+                max = Math.abs(array[i]);
+                normFlag = max > 1;
+            }
+        }
+
+        if (normFlag) {
+            for (int i = 0; i < array.length; i++) {
+                array[i] /= max;
+            }
+        }
+    }
+
+    private void correctForPitch(double[] stickSpeeds) {//x-tip
+        //if its over 45 there's no point in correcting it
+        if (Math.abs(pitchDegrees) < 45) {
+            //correctionFactor keeps the tilt correction within a certain threshold so it doesn't correct too much
+            double correctionFactor = (RobotMap.DRIVE.MAX_CORRECTION_FACTOR - RobotMap.DRIVE.MIN_CORRECTION_FACTOR) / (45 - RobotMap.DRIVE.PITCH_THRESHOLD_DEGREES);
```

Comment:
This smells of suspicious magic. I'll look into how it works later. (Not merge blocking, but copying formulas without understanding them is not really the best idea. Of course, I would _never_ do such a thing... /s)

Also, is this not constant? If it is, why not make it a `final` field of something (such as this very class or `RobotMap.DRIVE`)?